master/internal/usergroup/postgres_groups.go

```diff
@@ -217,79 +238,99 @@ func UpdateGroupTx(ctx context.Context, idb bun.IDB, group model.Group) error {
 
-// AddUsersToGroupTx adds users to a group by creating GroupMembership rows.
+// AddUsersToGroupsTx adds users to groups by creating GroupMembership rows.
 // Returns ErrNotFound if the group isn't found or ErrDuplicateRow if one
-// of the users is already in the group. Will use db.Bun() if passed nil
-// for idb.
-func AddUsersToGroupTx(ctx context.Context, idb bun.IDB, gid int, uids ...model.UserID) error {
+// of the users is already in the group (unless ignoreDuplicates).
+// Will use db.Bun() if passed nil for idb.
+func AddUsersToGroupsTx(ctx context.Context, idb bun.IDB, groups []int, ignoreDuplicates bool,
+	uids ...model.UserID,
+) error {
 	if idb == nil {
 		idb = db.Bun()
 	}
-	if _, err := GroupByIDTx(ctx, idb, gid); err != nil {
+
+	if err := ModifiableGroupsTx(ctx, idb, groups); err != nil {
 		return err
 	}
 
 	if len(uids) < 1 {
 		return nil
 	}
 
-	groupMem := make([]model.GroupMembership, 0, len(uids))
+	groupMem := make([]model.GroupMembership, 0, len(uids)*len(groups))
 	for _, uid := range uids {
-		groupMem = append(groupMem, model.GroupMembership{
-			UserID:  uid,
-			GroupID: gid,
-		})
+		for _, gid := range groups {
+			groupMem = append(groupMem, model.GroupMembership{
+				UserID:  uid,
+				GroupID: gid,
+			})
+		}
 	}
 
-	res, err := idb.NewInsert().Model(&groupMem).Exec(ctx)
-	if foundErr := db.MustHaveAffectedRows(res, err); foundErr != nil {
-		sError := db.MatchSentinelError(foundErr)
-		if errors.Is(sError, db.ErrNotFound) {
-			return errors.Wrapf(sError,
-				"Error adding %d user(s) to group %d because"+
-					" one or more of them were not found", len(uids), gid)
+	query := idb.NewInsert().Model(&groupMem)
+	if ignoreDuplicates {
+		query = query.On("CONFLICT(user_id, group_id) DO NOTHING")
+		_, err := query.Exec(ctx)
+		if err != nil {
+			return errors.Wrapf(err,
+				"Error adding %d user(s) to %d group(s)", len(uids), len(groups))
+		}
+	} else {
+		res, err := query.Exec(ctx)
+		if foundErr := db.MustHaveAffectedRows(res, err); foundErr != nil {
+			sError := db.MatchSentinelError(foundErr)
+			if errors.Is(sError, db.ErrNotFound) {
+				return errors.Wrapf(sError,
+					"Error adding %d user(s) to %d group(s) because"+
+						" one or more of them were not found", len(uids), len(groups))
+			}
+			return errors.Wrapf(sError, "Error when adding %d user(s) to %d group(s)",
+				len(uids), len(groups))
 		}
-		return errors.Wrapf(sError, "Error when adding %d user(s) to group %d",
-			len(uids), gid)
 	}
 
-	err = UpdateUsersTimestampTx(ctx, idb, uids)
+	err := UpdateUsersTimestampTx(ctx, idb, uids)
 	if err != nil {
 		return fmt.Errorf("error when updating users timestamps: %w", err)
 	}
 
 	return nil
 }
 
-// RemoveUsersFromGroupTx removes users from a group. Removes nothing and
-// returns ErrNotFound if the group or one of the users' membership rows
+// RemoveUsersFromGroupsTx removes users from a group. Removes nothing and
+// returns ErrNotFound if the group or all of the membership rows
```

Comment:
does this change behavior? 

It doesn't look like it but from the comment seems to imply one or more is a bug? I think it doesn't really matter too much either way

Reply:
no it doesn't change behavior, the previous error check was `db.MustHaveAffectedRows` and this unit test specifically allows a mix of member and non-member users (only change was putting testGroup in an array)

```go
err = RemoveUsersFromGroupsTx(ctx, nil, []int{testGroup.ID}, testUser.ID, -500)
require.NoError(t, err, 
    "erroneously returned error when trying to remove a mix of users in a group and not")
```